Investigate the `AssertionError` happening in the `shared_files` test

While running the `test_integrity_sync` test, using the tag `v4.3.0-rc1` it was seen the following error:
```
python3.9 -m pytest ../../test_cluster/test_integrity_sync/ -vvs
/home/yanazaeva/.local/lib/python3.9/site-packages/_testinfra_renamed.py:5: DeprecationWarning: testinfra package has been renamed to pytest-testinfra. Please `pip install pytest-testinfra` and `pip uninstall testinfra` and update your package requirements to avoid this message
  warnings.warn((
======================================= test session starts =======================================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.10.0, pluggy-1.0.0 -- /usr/bin/python3.9
cachedir: .pytest_cache
rootdir: /home/yanazaeva/git/wazuh-qa/tests/system
plugins: aiohttp-0.3.0, trio-0.7.0, testinfra-6.4.0, asyncio-0.16.0, cov-2.12.0, testinfra-6.0.0
collected 4 items                                                                                 

../../test_cluster/test_integrity_sync/test_integrity_sync.py::test_missing_file PASSED
../../test_cluster/test_integrity_sync/test_integrity_sync.py::test_shared_files FAILED
../../test_cluster/test_integrity_sync/test_integrity_sync.py::test_extra_files PASSED
../../test_cluster/test_integrity_sync/test_integrity_sync.py::test_extra_valid_files PASSED

============================================ FAILURES =============================================
________________________________________ test_shared_files ________________________________________

    def test_shared_files():
        """Check if the content of each file is the same in all nodes.
    
        Update the content of the files in the master node and check if they are updated in the workers.
        Then, update the content in the workers and check if it is overwritten by the one in the master.
        """
        agent_conf_content = '<agent_config></agent_config>'
        file_test_content_master = 'test_content_from_master'
        file_test_content_worker = 'test_content_from_worker'
    
        # Modify the content of each file in the master node to check if it is updated in the workers.
        for file in files_to_sync:
            host_manager.modify_file_content(host=test_hosts[0], path=file, content=file_test_content_master)
    
        # Modify the content of the agent.conf file to check if merged.mg file is updated in master and synchronized in
        # workers.
        host_manager.modify_file_content(host=test_hosts[0], path=agent_conf_file,
                                         content=f"{agent_conf_content}\n")
    
        time.sleep(time_to_sync)
    
        # Check whether files are correctly updated in the workers or not.
        for host in worker_hosts:
            for file in files_to_sync:
                result = host_manager.run_command(host, f'cat {file}')
                assert result == file_test_content_master, f'File {file} inside {host} should contain ' \
                                                           f'{file_test_content_master}, but it has: {result}'
    
        # Check whether the merged.mg file is correctly updated in master and synchronized in workers or not.
        for host in test_hosts:
            result = host_manager.run_command(host, f'cat {merged_mg_file}')
            # The agent.conf content will be before the !0 test_file line in merged.mg.
>           assert agent_conf_content in result, f'File {merged_mg_file} inside {host} should contain ' \
                                                 f'{agent_conf_content}, but it has: {result} '
E           AssertionError: File /var/ossec/etc/shared/test_group/merged.mg inside wazuh-worker1 should contain <agent_config></agent_config>, but it has: #test_group
E             !228 ar.conf
E             restart-ossec0 - restart-ossec.sh - 0
E             restart-ossec0 - restart-ossec.cmd - 0
E             restart-wazuh0 - restart-ossec.sh - 0
E             restart-wazuh0 - restart-ossec.cmd - 0
E             restart-wazuh0 - restart-wazuh - 0
E             restart-wazuh0 - restart-wazuh.exe - 0
E             !0 agent.conf
E             !0 test_file 
E           assert '<agent_config></agent_config>' in '#test_group\n!228 ar.conf\nrestart-ossec0 - restart-ossec.sh - 0\nrestart-ossec0 - restart-ossec.cmd - 0\nrestart-waz...ossec.cmd - 0\nrestart-wazuh0 - restart-wazuh - 0\nrestart-wazuh0 - restart-wazuh.exe - 0\n!0 agent.conf\n!0 test_file'

../../test_cluster/test_integrity_sync/test_integrity_sync.py:124: AssertionError
---------------------------------------- Captured log call ----------------------------------------
DEBUG    testinfra:base.py:286 RUN CommandResult(command=b"ansible --tree /tmp/tmp_kpjj85d -i /home/yanazaeva/git/wazuh-qa/tests/system/provisioning/agentless_cluster/inventory.yml -m copy --args 'src=/tmp/tmpaiw_lkhj dest=/var/ossec/etc/lists/test_file owner=wazuh group=wazuh mode=0775' wazuh-master", exit_status=0, stdout=b'wazuh-master | CHANGED => {\n    "changed": true,\n    "checksum": "8beb3373fcd7a0eaa4d8892146ba329c92ceea21",\n    "dest": "/var/ossec/etc/lists/test_file",\n    "gid": 999,\n    "group": "wazuh",\n    "md5sum": "a883275b0e4f51248b99a824ee28a356",\n    "mode": "0775",\n    "owner": "wazuh",\n    "size": 24,\n    "src": "/root/.ansible/tmp/ansible-tmp-1637838402.1333888-16136615941560/source",\n    "state": "file",\n    "uid": 999\n}\n', stderr=None)
INFO     testinfra:ansible.py:71 RUN Ansible('copy', 'src=/tmp/tmpaiw_lkhj dest=/var/ossec/etc/lists/test_file owner=wazuh group=wazuh mode=0775', {'check': False, 'become': False}): {'changed': True,
 'checksum': '8beb3373fcd7a0eaa4d8892146ba329c92ceea21',
 'dest': '/var/ossec/etc/lists/test_file',
 'gid': 999,
 'group': 'wazuh',
 'md5sum': 'a883275b0e4f51248b99a824ee28a356',
 'mode': '0775',
 'owner': 'wazuh',
 'size': 24,
 'src': '/root/.ansible/tmp/ansible-tmp-1637838402.1333888-16136615941560/source',
 'state': 'file',
 'uid': 999}
DEBUG    testinfra:base.py:286 RUN CommandResult(command=b"ansible --tree /tmp/tmpipk_t3m5 -i /home/yanazaeva/git/wazuh-qa/tests/system/provisioning/agentless_cluster/inventory.yml -m copy --args 'src=/tmp/tmpti5fugew dest=/var/ossec/etc/rules/test_file owner=wazuh group=wazuh mode=0775' wazuh-master", exit_status=0, stdout=b'wazuh-master | CHANGED => {\n    "changed": true,\n    "checksum": "8beb3373fcd7a0eaa4d8892146ba329c92ceea21",\n    "dest": "/var/ossec/etc/rules/test_file",\n    "gid": 999,\n    "group": "wazuh",\n    "md5sum": "a883275b0e4f51248b99a824ee28a356",\n    "mode": "0775",\n    "owner": "wazuh",\n    "size": 24,\n    "src": "/root/.ansible/tmp/ansible-tmp-1637838403.5622828-104297842951229/source",\n    "state": "file",\n    "uid": 999\n}\n', stderr=None)
INFO     testinfra:ansible.py:71 RUN Ansible('copy', 'src=/tmp/tmpti5fugew dest=/var/ossec/etc/rules/test_file owner=wazuh group=wazuh mode=0775', {'check': False, 'become': False}): {'changed': True,
 'checksum': '8beb3373fcd7a0eaa4d8892146ba329c92ceea21',
 'dest': '/var/ossec/etc/rules/test_file',
 'gid': 999,
 'group': 'wazuh',
 'md5sum': 'a883275b0e4f51248b99a824ee28a356',
 'mode': '0775',
 'owner': 'wazuh',
 'size': 24,
 'src': '/root/.ansible/tmp/ansible-tmp-1637838403.5622828-104297842951229/source',
 'state': 'file',
 'uid': 999}
DEBUG    testinfra:base.py:286 RUN CommandResult(command=b"ansible --tree /tmp/tmpbpr7ypdl -i /home/yanazaeva/git/wazuh-qa/tests/system/provisioning/agentless_cluster/inventory.yml -m copy --args 'src=/tmp/tmpyl6zsg9v dest=/var/ossec/etc/decoders/test_file owner=wazuh group=wazuh mode=0775' wazuh-master", exit_status=0, stdout=b'wazuh-master | CHANGED => {\n    "changed": true,\n    "checksum": "8beb3373fcd7a0eaa4d8892146ba329c92ceea21",\n    "dest": "/var/ossec/etc/decoders/test_file",\n    "gid": 999,\n    "group": "wazuh",\n    "md5sum": "a883275b0e4f51248b99a824ee28a356",\n    "mode": "0775",\n    "owner": "wazuh",\n    "size": 24,\n    "src": "/root/.ansible/tmp/ansible-tmp-1637838405.022413-139349899343646/source",\n    "state": "file",\n    "uid": 999\n}\n', stderr=None)
INFO     testinfra:ansible.py:71 RUN Ansible('copy', 'src=/tmp/tmpyl6zsg9v dest=/var/ossec/etc/decoders/test_file owner=wazuh group=wazuh mode=0775', {'check': False, 'become': False}): {'changed': True,
 'checksum': '8beb3373fcd7a0eaa4d8892146ba329c92ceea21',
 'dest': '/var/ossec/etc/decoders/test_file',
 'gid': 999,
 'group': 'wazuh',
 'md5sum': 'a883275b0e4f51248b99a824ee28a356',
 'mode': '0775',
 'owner': 'wazuh',
 'size': 24,
 'src': '/root/.ansible/tmp/ansible-tmp-1637838405.022413-139349899343646/source',
 'state': 'file',
 'uid': 999}
DEBUG    testinfra:base.py:286 RUN CommandResult(command=b"ansible --tree /tmp/tmp35sbas2e -i /home/yanazaeva/git/wazuh-qa/tests/system/provisioning/agentless_cluster/inventory.yml -m copy --args 'src=/tmp/tmp_ysqg0um dest=/var/ossec/var/multigroups/test_dir/merged.mg owner=wazuh group=wazuh mode=0775' wazuh-master", exit_status=0, stdout=b'wazuh-master | CHANGED => {\n    "changed": true,\n    "checksum": "8beb3373fcd7a0eaa4d8892146ba329c92ceea21",\n    "dest": "/var/ossec/var/multigroups/test_dir/merged.mg",\n    "gid": 999,\n    "group": "wazuh",\n    "md5sum": "a883275b0e4f51248b99a824ee28a356",\n    "mode": "0775",\n    "owner": "wazuh",\n    "size": 24,\n    "src": "/root/.ansible/tmp/ansible-tmp-1637838406.5366995-122652434875916/source",\n    "state": "file",\n    "uid": 999\n}\n', stderr=None)
INFO     testinfra:ansible.py:71 RUN Ansible('copy', 'src=/tmp/tmp_ysqg0um dest=/var/ossec/var/multigroups/test_dir/merged.mg owner=wazuh group=wazuh mode=0775', {'check': False, 'become': False}): {'changed': True,
 'checksum': '8beb3373fcd7a0eaa4d8892146ba329c92ceea21',
 'dest': '/var/ossec/var/multigroups/test_dir/merged.mg',
 'gid': 999,
 'group': 'wazuh',
 'md5sum': 'a883275b0e4f51248b99a824ee28a356',
 'mode': '0775',
 'owner': 'wazuh',
 'size': 24,
 'src': '/root/.ansible/tmp/ansible-tmp-1637838406.5366995-122652434875916/source',
 'state': 'file',
 'uid': 999}
DEBUG    testinfra:base.py:286 RUN CommandResult(command=b"ansible --tree /tmp/tmplf937ddy -i /home/yanazaeva/git/wazuh-qa/tests/system/provisioning/agentless_cluster/inventory.yml -m copy --args 'src=/tmp/tmplmh4g5fp dest=/var/ossec/etc/shared/test_group/test_file owner=wazuh group=wazuh mode=0775' wazuh-master", exit_status=0, stdout=b'wazuh-master | CHANGED => {\n    "changed": true,\n    "checksum": "8beb3373fcd7a0eaa4d8892146ba329c92ceea21",\n    "dest": "/var/ossec/etc/shared/test_group/test_file",\n    "gid": 999,\n    "group": "wazuh",\n    "md5sum": "a883275b0e4f51248b99a824ee28a356",\n    "mode": "0775",\n    "owner": "wazuh",\n    "size": 24,\n    "src": "/root/.ansible/tmp/ansible-tmp-1637838408.0339665-260610360752550/source",\n    "state": "file",\n    "uid": 999\n}\n', stderr=None)
INFO     testinfra:ansible.py:71 RUN Ansible('copy', 'src=/tmp/tmplmh4g5fp dest=/var/ossec/etc/shared/test_group/test_file owner=wazuh group=wazuh mode=0775', {'check': False, 'become': False}): {'changed': True,
 'checksum': '8beb3373fcd7a0eaa4d8892146ba329c92ceea21',
 'dest': '/var/ossec/etc/shared/test_group/test_file',
 'gid': 999,
 'group': 'wazuh',
 'md5sum': 'a883275b0e4f51248b99a824ee28a356',
 'mode': '0775',
 'owner': 'wazuh',
 'size': 24,
 'src': '/root/.ansible/tmp/ansible-tmp-1637838408.0339665-260610360752550/source',
 'state': 'file',
 'uid': 999}
DEBUG    testinfra:base.py:286 RUN CommandResult(command=b"ansible --tree /tmp/tmpwg6kc56v -i /home/yanazaeva/git/wazuh-qa/tests/system/provisioning/agentless_cluster/inventory.yml -m copy --args 'src=/tmp/tmpf3v18yxg dest=/var/ossec/etc/shared/test_group/agent.conf owner=wazuh group=wazuh mode=0775' wazuh-master", exit_status=0, stdout=b'wazuh-master | CHANGED => {\n    "changed": true,\n    "checksum": "bde5e153823b6718b5428686060569413bf13d6f",\n    "dest": "/var/ossec/etc/shared/test_group/agent.conf",\n    "gid": 999,\n    "group": "wazuh",\n    "md5sum": "d888c61e7ef75a766b4e9597c93c36fd",\n    "mode": "0775",\n    "owner": "wazuh",\n    "size": 30,\n    "src": "/root/.ansible/tmp/ansible-tmp-1637838409.514759-222600072518492/source",\n    "state": "file",\n    "uid": 999\n}\n', stderr=None)
INFO     testinfra:ansible.py:71 RUN Ansible('copy', 'src=/tmp/tmpf3v18yxg dest=/var/ossec/etc/shared/test_group/agent.conf owner=wazuh group=wazuh mode=0775', {'check': False, 'become': False}): {'changed': True,
 'checksum': 'bde5e153823b6718b5428686060569413bf13d6f',
 'dest': '/var/ossec/etc/shared/test_group/agent.conf',
 'gid': 999,
 'group': 'wazuh',
 'md5sum': 'd888c61e7ef75a766b4e9597c93c36fd',
 'mode': '0775',
 'owner': 'wazuh',
 'size': 30,
 'src': '/root/.ansible/tmp/ansible-tmp-1637838409.514759-222600072518492/source',
 'state': 'file',
 'uid': 999}
DEBUG    testinfra:base.py:286 RUN CommandResult(command=b"ansible --tree /tmp/tmp8ifa35lq -i /home/yanazaeva/git/wazuh-qa/tests/system/provisioning/agentless_cluster/inventory.yml -m command --args 'cat /var/ossec/etc/lists/test_file' wazuh-worker1", exit_status=0, stdout=b'wazuh-worker1 | CHANGED | rc=0 >>\ntest_content_from_master\n', stderr=None)
INFO     testinfra:ansible.py:71 RUN Ansible('command', 'cat /var/ossec/etc/lists/test_file', {'check': False, 'become': False}): {'changed': True,
 'cmd': ['cat', '/var/ossec/etc/lists/test_file'],
 'delta': '0:00:00.070688',
 'end': '2021-11-25 12:07:01.279050',
 'rc': 0,
 'start': '2021-11-25 12:07:01.208362',
 'stderr': '',
 'stderr_lines': [],
 'stdout': 'test_content_from_master',
 'stdout_lines': ['test_content_from_master']}
DEBUG    testinfra:base.py:286 RUN CommandResult(command=b"ansible --tree /tmp/tmpc4m1h5o6 -i /home/yanazaeva/git/wazuh-qa/tests/system/provisioning/agentless_cluster/inventory.yml -m command --args 'cat /var/ossec/etc/rules/test_file' wazuh-worker1", exit_status=0, stdout=b'wazuh-worker1 | CHANGED | rc=0 >>\ntest_content_from_master\n', stderr=None)
INFO     testinfra:ansible.py:71 RUN Ansible('command', 'cat /var/ossec/etc/rules/test_file', {'check': False, 'become': False}): {'changed': True,
 'cmd': ['cat', '/var/ossec/etc/rules/test_file'],
 'delta': '0:00:00.071131',
 'end': '2021-11-25 12:07:01.979481',
 'rc': 0,
 'start': '2021-11-25 12:07:01.908350',
 'stderr': '',
 'stderr_lines': [],
 'stdout': 'test_content_from_master',
 'stdout_lines': ['test_content_from_master']}
DEBUG    testinfra:base.py:286 RUN CommandResult(command=b"ansible --tree /tmp/tmpjl2lqf80 -i /home/yanazaeva/git/wazuh-qa/tests/system/provisioning/agentless_cluster/inventory.yml -m command --args 'cat /var/ossec/etc/decoders/test_file' wazuh-worker1", exit_status=0, stdout=b'wazuh-worker1 | CHANGED | rc=0 >>\ntest_content_from_master\n', stderr=None)
INFO     testinfra:ansible.py:71 RUN Ansible('command', 'cat /var/ossec/etc/decoders/test_file', {'check': False, 'become': False}): {'changed': True,
 'cmd': ['cat', '/var/ossec/etc/decoders/test_file'],
 'delta': '0:00:00.071407',
 'end': '2021-11-25 12:07:02.676203',
 'rc': 0,
 'start': '2021-11-25 12:07:02.604796',
 'stderr': '',
 'stderr_lines': [],
 'stdout': 'test_content_from_master',
 'stdout_lines': ['test_content_from_master']}
DEBUG    testinfra:base.py:286 RUN CommandResult(command=b"ansible --tree /tmp/tmpsr8y342z -i /home/yanazaeva/git/wazuh-qa/tests/system/provisioning/agentless_cluster/inventory.yml -m command --args 'cat /var/ossec/var/multigroups/test_dir/merged.mg' wazuh-worker1", exit_status=0, stdout=b'wazuh-worker1 | CHANGED | rc=0 >>\ntest_content_from_master\n', stderr=None)
INFO     testinfra:ansible.py:71 RUN Ansible('command', 'cat /var/ossec/var/multigroups/test_dir/merged.mg', {'check': False, 'become': False}): {'changed': True,
 'cmd': ['cat', '/var/ossec/var/multigroups/test_dir/merged.mg'],
 'delta': '0:00:00.069923',
 'end': '2021-11-25 12:07:03.382207',
 'rc': 0,
 'start': '2021-11-25 12:07:03.312284',
 'stderr': '',
 'stderr_lines': [],
 'stdout': 'test_content_from_master',
 'stdout_lines': ['test_content_from_master']}
DEBUG    testinfra:base.py:286 RUN CommandResult(command=b"ansible --tree /tmp/tmpl2h0t9xl -i /home/yanazaeva/git/wazuh-qa/tests/system/provisioning/agentless_cluster/inventory.yml -m command --args 'cat /var/ossec/etc/shared/test_group/test_file' wazuh-worker1", exit_status=0, stdout=b'wazuh-worker1 | CHANGED | rc=0 >>\ntest_content_from_master\n', stderr=None)
INFO     testinfra:ansible.py:71 RUN Ansible('command', 'cat /var/ossec/etc/shared/test_group/test_file', {'check': False, 'become': False}): {'changed': True,
 'cmd': ['cat', '/var/ossec/etc/shared/test_group/test_file'],
 'delta': '0:00:00.070166',
 'end': '2021-11-25 12:07:04.150433',
 'rc': 0,
 'start': '2021-11-25 12:07:04.080267',
 'stderr': '',
 'stderr_lines': [],
 'stdout': 'test_content_from_master',
 'stdout_lines': ['test_content_from_master']}
DEBUG    testinfra:base.py:286 RUN CommandResult(command=b"ansible --tree /tmp/tmpkwqpk8nu -i /home/yanazaeva/git/wazuh-qa/tests/system/provisioning/agentless_cluster/inventory.yml -m command --args 'cat /var/ossec/etc/lists/test_file' wazuh-worker2", exit_status=0, stdout=b'wazuh-worker2 | CHANGED | rc=0 >>\ntest_content_from_master\n', stderr=None)
INFO     testinfra:ansible.py:71 RUN Ansible('command', 'cat /var/ossec/etc/lists/test_file', {'check': False, 'become': False}): {'changed': True,
 'cmd': ['cat', '/var/ossec/etc/lists/test_file'],
 'delta': '0:00:00.071954',
 'end': '2021-11-25 12:07:04.896258',
 'rc': 0,
 'start': '2021-11-25 12:07:04.824304',
 'stderr': '',
 'stderr_lines': [],
 'stdout': 'test_content_from_master',
 'stdout_lines': ['test_content_from_master']}
DEBUG    testinfra:base.py:286 RUN CommandResult(command=b"ansible --tree /tmp/tmpmwtoiro5 -i /home/yanazaeva/git/wazuh-qa/tests/system/provisioning/agentless_cluster/inventory.yml -m command --args 'cat /var/ossec/etc/rules/test_file' wazuh-worker2", exit_status=0, stdout=b'wazuh-worker2 | CHANGED | rc=0 >>\ntest_content_from_master\n', stderr=None)
INFO     testinfra:ansible.py:71 RUN Ansible('command', 'cat /var/ossec/etc/rules/test_file', {'check': False, 'become': False}): {'changed': True,
 'cmd': ['cat', '/var/ossec/etc/rules/test_file'],
 'delta': '0:00:00.068205',
 'end': '2021-11-25 12:07:05.591028',
 'rc': 0,
 'start': '2021-11-25 12:07:05.522823',
 'stderr': '',
 'stderr_lines': [],
 'stdout': 'test_content_from_master',
 'stdout_lines': ['test_content_from_master']}
DEBUG    testinfra:base.py:286 RUN CommandResult(command=b"ansible --tree /tmp/tmp_1ody6dj -i /home/yanazaeva/git/wazuh-qa/tests/system/provisioning/agentless_cluster/inventory.yml -m command --args 'cat /var/ossec/etc/decoders/test_file' wazuh-worker2", exit_status=0, stdout=b'wazuh-worker2 | CHANGED | rc=0 >>\ntest_content_from_master\n', stderr=None)
INFO     testinfra:ansible.py:71 RUN Ansible('command', 'cat /var/ossec/etc/decoders/test_file', {'check': False, 'become': False}): {'changed': True,
 'cmd': ['cat', '/var/ossec/etc/decoders/test_file'],
 'delta': '0:00:00.072449',
 'end': '2021-11-25 12:07:06.298952',
 'rc': 0,
 'start': '2021-11-25 12:07:06.226503',
 'stderr': '',
 'stderr_lines': [],
 'stdout': 'test_content_from_master',
 'stdout_lines': ['test_content_from_master']}
DEBUG    testinfra:base.py:286 RUN CommandResult(command=b"ansible --tree /tmp/tmpsle795vf -i /home/yanazaeva/git/wazuh-qa/tests/system/provisioning/agentless_cluster/inventory.yml -m command --args 'cat /var/ossec/var/multigroups/test_dir/merged.mg' wazuh-worker2", exit_status=0, stdout=b'wazuh-worker2 | CHANGED | rc=0 >>\ntest_content_from_master\n', stderr=None)
INFO     testinfra:ansible.py:71 RUN Ansible('command', 'cat /var/ossec/var/multigroups/test_dir/merged.mg', {'check': False, 'become': False}): {'changed': True,
 'cmd': ['cat', '/var/ossec/var/multigroups/test_dir/merged.mg'],
 'delta': '0:00:00.072569',
 'end': '2021-11-25 12:07:07.014455',
 'rc': 0,
 'start': '2021-11-25 12:07:06.941886',
 'stderr': '',
 'stderr_lines': [],
 'stdout': 'test_content_from_master',
 'stdout_lines': ['test_content_from_master']}
DEBUG    testinfra:base.py:286 RUN CommandResult(command=b"ansible --tree /tmp/tmpkqu1bkms -i /home/yanazaeva/git/wazuh-qa/tests/system/provisioning/agentless_cluster/inventory.yml -m command --args 'cat /var/ossec/etc/shared/test_group/test_file' wazuh-worker2", exit_status=0, stdout=b'wazuh-worker2 | CHANGED | rc=0 >>\ntest_content_from_master\n', stderr=None)
INFO     testinfra:ansible.py:71 RUN Ansible('command', 'cat /var/ossec/etc/shared/test_group/test_file', {'check': False, 'become': False}): {'changed': True,
 'cmd': ['cat', '/var/ossec/etc/shared/test_group/test_file'],
 'delta': '0:00:00.068905',
 'end': '2021-11-25 12:07:07.754915',
 'rc': 0,
 'start': '2021-11-25 12:07:07.686010',
 'stderr': '',
 'stderr_lines': [],
 'stdout': 'test_content_from_master',
 'stdout_lines': ['test_content_from_master']}
DEBUG    testinfra:base.py:286 RUN CommandResult(command=b"ansible --tree /tmp/tmpurm0igr5 -i /home/yanazaeva/git/wazuh-qa/tests/system/provisioning/agentless_cluster/inventory.yml -m command --args 'cat /var/ossec/etc/shared/test_group/merged.mg' wazuh-master", exit_status=0, stdout=b'wazuh-master | CHANGED | rc=0 >>\n#test_group\n!228 ar.conf\nrestart-ossec0 - restart-ossec.sh - 0\nrestart-ossec0 - restart-ossec.cmd - 0\nrestart-wazuh0 - restart-ossec.sh - 0\nrestart-wazuh0 - restart-ossec.cmd - 0\nrestart-wazuh0 - restart-wazuh - 0\nrestart-wazuh0 - restart-wazuh.exe - 0\n!30 agent.conf\n<agent_config></agent_config>\n!24 test_file\ntest_content_from_master\n', stderr=None)
INFO     testinfra:ansible.py:71 RUN Ansible('command', 'cat /var/ossec/etc/shared/test_group/merged.mg', {'check': False, 'become': False}): {'changed': True,
 'cmd': ['cat', '/var/ossec/etc/shared/test_group/merged.mg'],
 'delta': '0:00:00.070503',
 'end': '2021-11-25 12:07:08.455490',
 'rc': 0,
 'start': '2021-11-25 12:07:08.384987',
 'stderr': '',
 'stderr_lines': [],
 'stdout': '#test_group\n'
           '!228 ar.conf\n'
           'restart-ossec0 - restart-ossec.sh - 0\n'
           'restart-ossec0 - restart-ossec.cmd - 0\n'
           'restart-wazuh0 - restart-ossec.sh - 0\n'
           'restart-wazuh0 - restart-ossec.cmd - 0\n'
           'restart-wazuh0 - restart-wazuh - 0\n'
           'restart-wazuh0 - restart-wazuh.exe - 0\n'
           '!30 agent.conf\n'
           '<agent_config></agent_config>\n'
           '!24 test_file\n'
           'test_content_from_master',
 'stdout_lines': ['#test_group',
                  '!228 ar.conf',
                  'restart-ossec0 - restart-ossec.sh - 0',
                  'restart-ossec0 - restart-ossec.cmd - 0',
                  'restart-wazuh0 - restart-ossec.sh - 0',
                  'restart-wazuh0 - restart-ossec.cmd - 0',
                  'restart-wazuh0 - restart-wazuh - 0',
                  'restart-wazuh0 - restart-wazuh.exe - 0',
                  '!30 agent.conf',
                  '<agent_config></agent_config>',
                  '!24 test_file',
                  'test_content_from_master']}
DEBUG    testinfra:base.py:286 RUN CommandResult(command=b"ansible --tree /tmp/tmp_690jkxf -i /home/yanazaeva/git/wazuh-qa/tests/system/provisioning/agentless_cluster/inventory.yml -m command --args 'cat /var/ossec/etc/shared/test_group/merged.mg' wazuh-worker1", exit_status=0, stdout=b'wazuh-worker1 | CHANGED | rc=0 >>\n#test_group\n!228 ar.conf\nrestart-ossec0 - restart-ossec.sh - 0\nrestart-ossec0 - restart-ossec.cmd - 0\nrestart-wazuh0 - restart-ossec.sh - 0\nrestart-wazuh0 - restart-ossec.cmd - 0\nrestart-wazuh0 - restart-wazuh - 0\nrestart-wazuh0 - restart-wazuh.exe - 0\n!0 agent.conf\n!0 test_file\n', stderr=None)
INFO     testinfra:ansible.py:71 RUN Ansible('command', 'cat /var/ossec/etc/shared/test_group/merged.mg', {'check': False, 'become': False}): {'changed': True,
 'cmd': ['cat', '/var/ossec/etc/shared/test_group/merged.mg'],
 'delta': '0:00:00.068337',
 'end': '2021-11-25 12:07:09.155458',
 'rc': 0,
 'start': '2021-11-25 12:07:09.087121',
 'stderr': '',
 'stderr_lines': [],
 'stdout': '#test_group\n'
           '!228 ar.conf\n'
           'restart-ossec0 - restart-ossec.sh - 0\n'
           'restart-ossec0 - restart-ossec.cmd - 0\n'
           'restart-wazuh0 - restart-ossec.sh - 0\n'
           'restart-wazuh0 - restart-ossec.cmd - 0\n'
           'restart-wazuh0 - restart-wazuh - 0\n'
           'restart-wazuh0 - restart-wazuh.exe - 0\n'
           '!0 agent.conf\n'
           '!0 test_file',
 'stdout_lines': ['#test_group',
                  '!228 ar.conf',
                  'restart-ossec0 - restart-ossec.sh - 0',
                  'restart-ossec0 - restart-ossec.cmd - 0',
                  'restart-wazuh0 - restart-ossec.sh - 0',
                  'restart-wazuh0 - restart-ossec.cmd - 0',
                  'restart-wazuh0 - restart-wazuh - 0',
                  'restart-wazuh0 - restart-wazuh.exe - 0',
                  '!0 agent.conf',
                  '!0 test_file']}
===================================== short test summary info =====================================
FAILED ../../test_cluster/test_integrity_sync/test_integrity_sync.py::test_shared_files - Assert...
============================= 1 failed, 3 passed in 232.73s (0:03:52) =============================
``` 

This problem does not take place every time you run the test, as sometimes the output is the desired one:
```
python3.9 -m pytest ../../test_cluster/test_integrity_sync/ -vvs
/home/yanazaeva/.local/lib/python3.9/site-packages/_testinfra_renamed.py:5: DeprecationWarning: testinfra package has been renamed to pytest-testinfra. Please `pip install pytest-testinfra` and `pip uninstall testinfra` and update your package requirements to avoid this message
  warnings.warn((
========================================================== test session starts ===========================================================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.10.0, pluggy-1.0.0 -- /usr/bin/python3.9
cachedir: .pytest_cache
rootdir: /home/yanazaeva/git/wazuh-qa/tests/system
plugins: aiohttp-0.3.0, trio-0.7.0, testinfra-6.4.0, asyncio-0.16.0, cov-2.12.0, testinfra-6.0.0
collected 4 items                                                                                                                        

../../test_cluster/test_integrity_sync/test_integrity_sync.py::test_missing_file PASSED
../../test_cluster/test_integrity_sync/test_integrity_sync.py::test_shared_files PASSED
../../test_cluster/test_integrity_sync/test_integrity_sync.py::test_extra_files PASSED
../../test_cluster/test_integrity_sync/test_integrity_sync.py::test_extra_valid_files PASSED

===================================================== 4 passed in 255.43s (0:04:15) ======================================================
```

Due to this, the possible reason behind this behavior could be a race condition while updating the `merged.mg` file.

--- a/tests/system/test_cluster/test_integrity_sync/test_integrity_sync.py
+++ b/tests/system/test_cluster/test_integrity_sync/test_integrity_sync.py
@@ -16,7 +16,7 @@ worker_hosts = test_hosts[1:]
 inventory_path = os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))),
                               'provisioning', 'agentless_cluster', 'inventory.yml')
 
-time_to_sync = 10
+time_to_sync = 21
 host_manager = HostManager(inventory_path)
 client_keys_path = os.path.join(WAZUH_PATH, "etc", "client.keys")
 

